libs/common/src/platform/services/config/default-config.service.ts

```diff
@@ -90,19 +90,20 @@ export class DefaultConfigService implements ConfigService {
       map((config) => config?.environment?.cloudRegion ?? Region.US),
     );
   }
-  getFeatureFlag$<T extends FeatureFlagValue>(key: FeatureFlag, defaultValue?: T) {
+
+  getFeatureFlag$<Flag extends FeatureFlag>(key: Flag, defaultValue?: FeatureFlagType<Flag>) {
     return this.serverConfig$.pipe(
       map((serverConfig) => {
         if (serverConfig?.featureStates == null || serverConfig.featureStates[key] == null) {
           return defaultValue;
         }
 
-        return serverConfig.featureStates[key] as T;
+        return serverConfig.featureStates[key] as FeatureFlagType<Flag>;
```

Comment:
:art: suggestion: Since we now have the types available during runtime, maybe we should add some guards here to verify that we actually get the types we expect?

```suggestion
        const value = serverConfig.featureStates[key];
        if (FeatureFlagRuntimeTypes[key] !== typeof value) { throw new Error("Unknown flag value, possibly undefined?"); } // or maybe return default value?
        return value as FeatureFlagType<Flag>;
```

Or maybe this check should be done when fetching the server config, in which case `serverConfig?.featureStates` might already be fully type-safe by this point

Reply:
I was planning on using it in my other PR.